cmake/Utils.cmake

```
@@ -15,7 +15,7 @@ endmacro()
 ## TODO: Add vsim target or some such
 macro(add_chimera_test name)
   add_chimera_executable(${ARGV})
-  add_test(NAME ${name} COMMAND ${name})
+  add_test(NAME ${name} COMMAND ${SIMULATION_BINARY} +BINARY=$<TARGET_FILE:${name}> +PRELMODE=${PRELOAD_MODE_INT})
```

This is very specific to simulation with `vsim` but would break with `GVSoC`. Why not populating a `SIMULATION_FLAGS` variable based on the simulation platform (e.g. `rtl`, `gvsoc`) in the top-level CMakeLists.txt?

```suggestion
  add_test(NAME ${name} COMMAND ${SIMULATION_BINARY} ${SIMULATION_FLAGS}
```

---

`simulation` is not meant to include `GVSoC` - `simulation` should only include RTL simulators, no SoC emulators. This invocation is also not specific to `vsim` - it's specific to the [main testbench of chimera](https://github.com/pulp-platform/chimera/blob/devel/target/sim/src/tb_chimera_soc.sv#L89) and should work with any simulator that models this testbench (through the use of a binary - e.g. vcs or verilator).

I'm not opposed to GVSoC support, but abstracting it in here seems forced. I'd suggest to add a switch case once we get to adding GVSoC support.